### PR TITLE
MTDSA-4308 Added connector, modified request obj and parser

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: app/v1/connectors/BaseDesConnector.scala [hunk 1]
```diff
@@ -56,4 +56,16 @@ trait BaseDesConnector {
     doGet(desHeaderCarrier(hc))
   }
 
+  def get[Resp](uri: DesUri[Resp], param: Seq[(String, String)])(implicit ec: ExecutionContext,
+                                                                 hc: HeaderCarrier,
+                                                                 httpReads: HttpReads[DesOutcome[Resp]]): Future[DesOutcome[Resp]] = {
+
+    def doGet(implicit hc: HeaderCarrier): Future[DesOutcome[Resp]] = {
+      http.GET(s"${appConfig.desBaseUrl}/${uri.value}", queryParams = param)
```
Comment (BetaDraconis) [inline app/v1/connectors/BaseDesConnector.scala]:
```suggestion
      http.GET(s"${appConfig.desBaseUrl}/${uri.value}", queryParams)
```

File: app/v1/connectors/BaseDesConnector.scala [hunk 2]
```diff
@@ -56,4 +56,16 @@ trait BaseDesConnector {
     doGet(desHeaderCarrier(hc))
   }
 
+  def get[Resp](uri: DesUri[Resp], param: Seq[(String, String)])(implicit ec: ExecutionContext,
```
Comment (BetaDraconis) [inline app/v1/connectors/BaseDesConnector.scala]:
```suggestion
  def get[Resp](uri: DesUri[Resp], queryParams: Seq[(String, String)])(implicit ec: ExecutionContext,
```

Comment (BetaDraconis) [inline app/v1/connectors/BaseDesConnector.scala]: I think this also needs a test scenario in BaseDesConnectorSpec

File: app/utils/DateUtils.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+package utils
+
+import java.time.{LocalDateTime, Year}
+
+import v1.models.requestData.DesTaxYear
+
+object DateUtils {
+
+  def getDesTaxYear(taxYear: Option[String]): DesTaxYear = {
```
Comment (BetaDraconis) [inline app/utils/DateUtils.scala]: This needs unit testing

File: app/utils/DateUtils.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    taxYear.fold({
+      if(LocalDateTime.now().getMonthValue <= 3)
+        DesTaxYear(Year.now().toString)
```
Comment (james-work-account) [inline app/utils/DateUtils.scala]: Line not hit by coverage

**Pull request merged**